hoomd/md/IntegratorTwoStep.cc

```diff
@@ -478,6 +473,62 @@ void IntegratorTwoStep::setAutotunerParams(bool enable, unsigned int period)
         method->setAutotunerParams(enable, period);
     }
 
+/// helper function to compute net force/virial
+void IntegratorTwoStep::computeNetForce(uint64_t timestep)
+    {
+    if (m_rigid_bodies)
+        {
+        m_rigid_bodies->validateRigidBodies();
```

Comment:
I see that `validateRigidBodies` has an early exit condition. What conditions invalidate that? It is an expensive operation, so we wouldn't want to run it too often.

Reply:
A change in rigid body definitions or the addition or removal of particles so infrequent events in most simulations. Without this though, either of these could leave the simulation in an invalid state without an error potentially, The addition or removal of particles could lead to error or no error or potentially segfaults.

Reply:
OK. So this is only a performance penalty if the user dynamically changes their rigid bodies or if they dynamically add particles to the system. For this PR, let's document that these cases will degrade performance. Later, if any user has a strong need for such simulations we can relax it.  For example, only changing the rigid body positions doesn't require a full revalidation - only changing the numbers. New free particles added to the system doesn't need to trigger revalidation, but new rigid bodies does.